### PR TITLE
Feature/iep 617 date validation for progress report endpoint for liq03 api

### PR DESCRIPTION
**Pull request by adeharrison-ch**

### CODE REVIEW

**Review by CHssmith2 — COMMENTED** [review]
Looks good to me with enough test coverage.

File: models/request_entities.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type StatementOfAffairs struct {
 
 // ProgressReport is the model to represent a progress report for an insolvency case
 type ProgressReport struct {
-	FromDate    string   `json:"from_date"`
-	ToDate      string   `json:"to_date"`
+	FromDate    string   `json:"from_date" validate:"required,datetime=2006-01-02"`
+	ToDate      string   `json:"to_date" validate:"required,datetime=2006-01-02"`
```
Comment (Jideboris) [inline models/request_entities.go:60]: presumably these dates are from the requirement and would not change, right?

File: utils/date_test.go [hunk 2]
```diff
@@ -41,16 +124,16 @@ func TestUnitIsValidDate(t *testing.T) {
...
 		incorporatedOn := "2020-06-06"
 
-		isDateValid, err := IsValidDate(date, incorporatedOn)
+		isDateValid, err := IsDateBetweenIncorporationAndNow(date, incorporatedOn)
 		So(isDateValid, ShouldEqual, false)
 		So(err, ShouldEqual, nil)
 	})
 
-	Convey("valid date", t, func() {
+	Convey("Date is between Incorporation date and Now", t, func() {
```
Comment (Jideboris) [inline utils/date_test.go:132]: Date in this test is not now, perhaps we should rename the description

**Review by Jideboris — APPROVED** [review]

**Comment by Jideboris** [comment]
I suggest you get this PR squashed.

**Pull request merged**